server/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java

```diff
@@ -1695,131 +1596,235 @@ protected void setConfigProperties() throws ConfigurationException, IOException,
         log.debug("Old values: {}", oldValues);
     }
 
-    /**
-     * @see #doStartAndWait(boolean)
-     */
-    public boolean doStartAndWait() throws PackageException {
-        boolean started = doStartAndWait(false);
-        if (started && !quiet) {
-            log.info("Go to {}", this::getURL);
-        }
-        return started;
-    }
-
-    /**
-     * @see #stop(boolean)
-     */
-    public void stop() {
-        stop(false);
-    }
-
     /**
      * Call {@link #doStart(boolean)} with false as parameter.
      *
      * @see #doStart(boolean)
      * @return true if the server started successfully
      */
-    public boolean doStart() throws PackageException {
-        boolean started = doStart(false);
-        if (started && !quiet) {
-            log.info("Go to {}", this::getURL);
+    public boolean doStart() {
+        if (doStart(false).isAlive()) {
+            if (!quiet) {
+                log.info("Go to {}", this::getURL);
+            }
+            return true;
         }
-        return started;
+        return false;
     }
 
     /**
-     * Whereas {@link #doStart()} considers the server as started when the process is running, {@link #doStartAndWait()}
-     * waits for effective start by watching the logs
+     * Whereas {@link #doStart()} considers the server as started when the process is running, this method waits for
+     * effective start by calling {@link #statusServletClient}.
      *
-     * @param logProcessOutput Must process output stream must be logged or not.
      * @return true if the server started successfully
      */
-    public boolean doStartAndWait(boolean logProcessOutput) throws PackageException {
-        boolean commandSucceeded = false;
-        if (doStart(logProcessOutput)) {
-            addShutdownHook();
-            try {
-                if (waitForEffectiveStart()) {
-                    commandSucceeded = true;
+    public boolean doStartAndWait() {
+        var nuxeoProcess = doStart(false);
+        if (nuxeoProcess.isAlive()) {
+            //noinspection unused
+            try (var hook = new ShutdownHook(this)) { // NOSONAR try-with-resources
```

Comment:
is `// NOSONAR try-with-resources` really useful?

Reply:
yes, otherwise we get a warning about unused `hook` variable

Reply:
OK, not in Eclipse :)

Reply:
For Idea it's the `//noinspection unused` which does that, what is the eclipse equivalent?

Reply:
NOSONAR removed